### PR TITLE
♻️ Move RTC manager to a service

### PR DESCRIPTION
**Pull request by powerivq**
Redo https://github.com/ampproject/amphtml/pull/32386

https://github.com/ampproject/amphtml/issues/32713 Ran into this issue, and this PR has it fixed. The root cause is that different amp-ad elements are incorrectly sharing the same service instance.

### CODE REVIEW

**Review by rcebulko — COMMENTED** [review]
Changes to `services.js` look fine to be, but could you loop in another reviewer who's more familiar with the rest of this code before I sign off? I don't feel confident reviewing this fully.

**Comment by powerivq** [comment]
@rcebulko sorry, I just forgot to add monetization reviewers.

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 1]
```diff
@@ -2376,31 +2377,23 @@ export class AmpA4A extends AMP.BaseElement {
...
-      user().error(
-        TAG,
-        'RTC not supported for ad network ' +
-          `${this.element.getAttribute('type')}`
+    if (this.element.getAttribute('rtc-config')) {
+      installRealTimeConfigServiceForDoc(this.getAmpDoc());
+      return this.getBlockRtc_().then((shouldBlock) =>
+        shouldBlock
+          ? undefined
+          : Services.realTimeConfigForDoc(
```
Comment (calebcordry) [inline extensions/amp-a4a/0.1/amp-a4a.js:2385]: should we leave this wrapped in a try/catch?

File: extensions/amp-a4a/0.1/amp-a4a.js [hunk 2]
```diff
@@ -2376,31 +2377,23 @@ export class AmpA4A extends AMP.BaseElement {
...
-            consentState,
-            consentString,
-            consentMetadata,
-            this.verifyStillCurrent()
-          );
-        } catch (err) {
-          user().error(TAG, 'Could not perform Real Time Config.', err);
-        }
-      });
-    } else if (this.element.getAttribute('rtc-config')) {
```
Comment (calebcordry) [inline extensions/amp-a4a/0.1/amp-a4a.js:2399]: did we want to get rid of this warning?

**Review by calebcordry — APPROVED** [review]
Approving but perhaps we should consider wrapping this in an experiment if possible, since we already broke it once.

File: extensions/amp-a4a/0.1/amp-a4a.js (hunk 2 above)
Comment (powerivq) [reply extensions/amp-a4a/0.1/amp-a4a.js:2399]: Yes, this is intentional. In the past, we rely on using whether `AMP.RealTimeConfigManager` is present to determine if the ad network supports RTC, but this is the wrong criteria.

**Comment by powerivq** [comment]
@calebcordry RTV slots are full now. We can still try looking at the beta numbers (last time I didn't think it helped though).

**Review by rcebulko — APPROVED** [review]

**Pull request merged**